### PR TITLE
Enable perf try builder

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This adds a dedicated branch for perf to use for CI, intended to allow perf to
enqueue builds without needing to use bors. bors is great, but bors requires an
open PR to work, and we want to invoke perf on closed PRs sometimes (in
particular, rollups).

### CODE REVIEW

File: src/ci/github-actions/ci.yml [hunk 1]
```diff
@@ -261,7 +262,7 @@ jobs:
     name: try
     env:
       <<: [*shared-ci-variables, *prod-variables]
-    if: github.event_name == 'push' && github.ref == 'refs/heads/try' && github.repository == 'rust-lang-ci/rust'
+    if: github.event_name == 'push' && github.ref == 'refs/heads/try' && github.ref != 'refs/heads/try-perf' && github.repository == 'rust-lang-ci/rust'
```
Comment (pietroalbini) [inline src/ci/github-actions/ci.yml:265]: Why did you add this `if` condition?

`github.ref != 'refs/heads/try-perf'` is always going to be true if `github.ref == 'refs/heads/try'` is also true.

File: src/ci/github-actions/ci.yml [hunk 2]
```diff
@@ -89,7 +89,7 @@ x--expand-yaml-anchors--remove:
         uses: rust-lang/simpleinfra/github-actions/cancel-outdated-builds@master
         with:
           github_token: "${{ secrets.github_token }}"
-        if: success() && !env.SKIP_JOB && github.ref != 'refs/heads/try'
+        if: success() && !env.SKIP_JOB && github.ref != 'refs/heads/try' && github.ref != 'refs/heads/try-perf'
```
Comment (pietroalbini) [inline src/ci/github-actions/ci.yml:92]: I think we might actually want `cancel-outdated-builds` since we only plan to run one build at the time.

**Comment by Mark-Simulacrum** [comment]
r? @pietroalbini 

Fixed both nits.
